### PR TITLE
refactor org search

### PR DESCRIPTION
**Pull request by xgp**
updated how org search works. refactored to simplify to one method in provider.

deprecated other get/search stream methods.

### CODE REVIEW

**Comment by xgp** [comment]
@phamann Please take a look if you have a moment. I updated some of your predicate code in the `OrganizationProvider` as well as simplified the interface there (+deprecated a few methods). Would be great to have your eyes/feedback!

We also discovered (independently of your earlier update) some problems with fetching and paginating results when the `/orgs` query was coming from an org member user, rather than a user with `view-organizations` role that was also fixed by moving the logic up into the query criteria.

File: src/main/java/io/phasetwo/service/model/jpa/JpaOrganizationProvider.java [hunk 1]
```diff
@@ -249,9 +236,11 @@ private List<Predicate> predicates(
 
       switch (key) {
         case "name":
-          predicates.add(builder.like(root.get(key), "%" + value.toLowerCase() + "%"));
+          predicates.add(
+              builder.or(
+                  builder.like(root.get("name"), "%" + value.toLowerCase() + "%"),
+                  builder.like(root.get("displayName"), "%" + value.toLowerCase() + "%")));
```
Comment (phamann) [inline src/main/java/io/phasetwo/service/model/jpa/JpaOrganizationProvider.java:242]: Nice!

**Review by phamann — APPROVED** [review]
@xgp LGTM. I like the addition to search by display name which will resolve another of the recent feature requests.

**Comment by xgp** [comment]
@phamann thanks for taking the time to look. Really appreciate it!

**Pull request merged**